### PR TITLE
feat(ingest): Data domains ingestion

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/source_docs/hive.md [hunk 1]
```diff
@@ -108,26 +108,29 @@ Note that a `.` is used to denote nested fields in the YAML recipe.
...
+| `view_pattern.ignoreCase`       |          | `True`   | Whether to ignore case sensitivity during pattern matching.                                                                                                                             |
+| `include_tables`                |          | `True`   | Whether tables should be ingested.                                                                                                                                                      |
+| `domains.domain_key.allow`      |          |          | List of regex patterns for tables/schemas to set domain_key domain key (domain_key can be any string like `sales`. There can be multiple domain key specified.                          |
+| `domains.domain_key.deny`       |          |          | List of regex patterns for tables/schemas to not assign domain_key. There can be multiple domain key specified.                                                                         |
```
Comment (jjoyce0510) [inline metadata-ingestion/source_docs/hive.md:132]: Use case?

File: metadata-ingestion/source_docs/mssql.md [hunk 2]
```diff
@@ -81,29 +81,32 @@ Note that a `.` is used to denote nested fields in the YAML recipe.
...
+| `view_pattern.ignoreCase`       |          | `True`             | Whether to ignore case sensitivity during pattern matching.                                                                                                                             |
+| `include_tables`                |          | `True`             | Whether tables should be ingested.                                                                                                                                                      |
+| `include_views`                 |          | `True`             | Whether views should be ingested.                                                                                                                                                       |
+| `domains.domain_key.allow`      |          |                    | List of regex patterns for tables/schemas to set domain_key domain key (domain_key can be any string like `sales`. There can be multiple domain key specified.                          |
```
Comment (jjoyce0510) [inline metadata-ingestion/source_docs/mssql.md:107]: We should note specifically that domain_key can be an ID or an URN of a domain

File: metadata-ingestion/src/datahub/emitter/mcp_builder.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+from datahub.emitter.mce_builder import make_domain_urn
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.source import SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.domain import DomainProperties
+from datahub.metadata.schema_classes import ChangeTypeClass, DomainsClass
+
+
+def add_domain_wu(
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/emitter/mcp_builder.py:11]: Left comments on the other PR :)

File: metadata-ingestion/src/datahub/emitter/mce_builder.py [hunk 4]
```diff
@@ -130,6 +130,10 @@ def make_chart_urn(platform: str, name: str) -> str:
     return f"urn:li:chart:({platform},{name})"
 
 
+def make_domain_urn(domain: str) -> str:
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/emitter/mce_builder.py:133]: Left comment on other PR :)

File: metadata-ingestion/source_docs/bigquery.md [hunk 5]
```diff
@@ -134,6 +134,9 @@ As a SQL-based service, the Athena integration is also supported by our SQL prof
...
 | `bigquery_audit_metadata_datasets`     |                                                                           | None                                                                    | A list of datasets that contain a table named `cloudaudit_googleapis_com_data_access` which contain BigQuery audit logs, specifically, those containing `BigQueryAuditMetadata`. It is recommended that the project of the dataset is also specified, for example, `projectA.datasetB`. |
+| `domain.domain_key.allow`              |                                                                           |                                                                         | List of regex patterns for tables/schemas to set domain_key domain key (domain_key can be any string like `sales`. There can be multiple domain key specified.                                                                                                                          |
```
Comment (shirshanka) [inline metadata-ingestion/source_docs/bigquery.md:137]: What does it mean "There can be multiple domain key specified" -- Only one per entity right now, right?

File: metadata-ingestion/source_docs/bigquery.md [hunk 6]
```diff
@@ -134,6 +134,9 @@ As a SQL-based service, the Athena integration is also supported by our SQL prof
...
+| `domain.domain_key.deny`               |                                                                           |                                                                         | List of regex patterns for tables/schemas to not assign domain_key. There can be multiple domain key specified.                                                                                                                                                                         |
+| `domain.domain_key.ignoreCase`         |                                                                           | `True`                                                                  | Whether to ignore case sensitivity during pattern matching.There can be multiple domain key specified.                                                                                                                                                                                  |
```
Comment (jjoyce0510) [inline metadata-ingestion/source_docs/bigquery.md:139]: nit: missing a space after the period

**Pull request closed without merging**